reproducibility/codalab/reproduce_results.py

```diff
@@ -0,0 +1,436 @@
+from itertools import product
+
+from examples.configs.algorithm import algorithm_defaults
+from examples.configs.datasets import dataset_defaults
+from reproducibility.codalab.hyperparameter_search_space import (
+    ID_HYPERPARAMETER_SEARCH_SPACE,
+)
+from reproducibility.codalab.util.analysis_utils import (
+    compile_results,
+    get_early_stopped_row,
+    get_metrics,
+    load_results,
+)
+
+"""
+Reproduce results of the WILDS paper in CodaLab.
+
+[TODO's]
+- Generate results of paper
+- ID val ERM
+
+Example usage:
+
+    python reproducibility/codalab/reproduce_results.py 
+"""
+
+
+class CodaLabReproducibility:
+    _GROUP_NAME = "wilds-admins"
+    _ADDITIONAL_DATASETS = ["bdd100k", "celebA", "waterbirds", "yelp"]
+    _CIVIL_COMMENTS_ADDITIONAL_ALGORITHMS = [
+        "erm_groupby-y",
+        "erm_groupby-black-y",
+        "groupDRO_groupby-y",
+        "groupDRO_groupby-black-y",
+    ]
+
+    def __init__(self, wilds_version, worksheet_name, all_datasets=False):
+        self._wilds_version = wilds_version
+        self._worksheet_name = (
+            worksheet_name if worksheet_name else f"wilds-results-{wilds_version}"
+        )
+        self._all_datasets = all_datasets
+
+        # Run experiments on the main instance - https://worksheets.codalab.org
+        self._run(["cl", "work", "https://worksheets.codalab.org::"])
+
+    def reproduce_main_results(self):
+        worksheet_uuid = self._set_worksheet()
+        datasets_uuids = self._get_datasets_uuids(worksheet_uuid)
+        wilds_src_uuid = self._get_bundle_uuid("wilds", worksheet_uuid)
+
+        for algorithm in algorithm_defaults.keys():
+            self._add_header(algorithm)
+
+            for dataset in dataset_defaults.keys():
+                # Skip the additional datasets if we don't want to reproduce results for all datasets
+                if (
+                    not self._all_datasets
+                    and dataset in CodaLabReproducibility._ADDITIONAL_DATASETS
+                ) or (algorithm in ["IRM", "deepCORAL"] and dataset == "civilcomments"):
+                    continue
+
+                self._add_header(dataset, level=5)
+                dataset_fullname = f"{dataset}_{self._wilds_version}"
+                dataset_uuid = datasets_uuids[dataset]
+                if dataset == "poverty":
+                    folds = ["A", "B", "C", "D", "E"]
+                    for fold in folds:
+                        self._run(
+                            [
+                                "cl",
+                                "run",
+                                f"--name=${dataset}_${algorithm}_fold${fold}",
+                                "--request-docker-image=codalabrunner/wilds_test:0.1",  # TODO: hardcoded
+                                f"--request-cpus=4",
+                                "--request-gpus=1",
+                                "--request-disk=10g",
+                                "--request-memory=54g",
```

Comment:
Do you know if any of our jobs actually require this much memory? (Is there some way to see peak utilization within a cl bundle?

Reply:
This is a new feature that is coming soon: https://github.com/codalab/codalab-worksheets/pull/3272